client/signup/index.node.js

```diff
@@ -3,38 +3,19 @@
 /**
  * Internal dependencies
  */
+import { getLanguage, getLanguageSlugs } from 'lib/i18n-utils';
 
-import { getLanguage } from 'lib/i18n-utils';
+const lang = `:lang(${ getLanguageSlugs().join( '|' ) })?`;
 
 export default function( router ) {
-	router( '/start/:flowName?/:stepName?/:stepSectionName?/:lang?', setUpLocale );
+	router( `/start/:flowName?/:stepName?/:stepSectionName?/${ lang }`, setUpLocale );
 }
 
 // Set up the locale in case it has ended up in the flow param
 function setUpLocale( context, next ) {
-	let { flowName, stepName, stepSectionName, lang } = context.params;
-
-	if ( ! lang && stepSectionName && getLanguage( stepSectionName ) ) {
-		lang = stepSectionName;
-		stepSectionName = undefined;
-	} else if ( ! lang && stepName && getLanguage( stepName ) ) {
-		lang = stepName;
-		stepName = undefined;
-	} else if ( ! lang && flowName && getLanguage( flowName ) ) {
-		lang = flowName;
-		flowName = undefined;
```

Comment:
👍 to removing this block.

Not sure if this is imperative, but since the `getLanguage` check no longer runs over all params, we don't set the language after refreshing the page, for example, http://calypso.localhost:3000/start/user/it.

As far as I know this is edge-case territory, however just flagging it. 

Looks like login has array of [web routes](https://github.com/Automattic/wp-calypso/blob/e6df0ec/client/login/index.web.js#L33) with possible manifestations using `lang`. So I can, for example, successfully refresh http://calypso.localhost:3000/log-in/link/fr

Reply:
> Not sure if this is imperative, but since the `getLanguage` check no longer runs over all params, we don't set the language after refreshing the page, for example, http://calypso.localhost:3000/start/user/it.

I might not be following :sweat_smile: If I enter http://calypso.localhost:3000/start/user/it into an incognito window, I'm redirected to http://calypso.localhost:3000/start/about/it, but it's still in Italian. Seems like the desired behavior (as found in production) to me...? Or am I missing something? :smile:

Reply:
Ugh, I could have expressed that more clearly. A small regression, but now we ignore the locale parameter when refreshing the page in between steps.

Here's a video!

https://cldup.com/5pgvvZmUUh.mp4

But it doesn't affect the flow, only thought I'd mention it.

Reply:
Hmm, good spot. Apparently, due to the optional nature of most route fragments, the language ends up in `stepName` there. `console.log( context.params )` gives

```js
{ flowName: 'domains',
  stepName: 'it',
  stepSectionName: undefined,
  lang: undefined }
```

Reply:
I used a variation of @dmsnell's [suggestion](https://github.com/Automattic/wp-calypso/pull/25378#discussion_r195944214) to make all route fragments non-optional, which fixes this regression.

Reply:
oops - I didn't see all the corners before I left my last comment. sorry for droning on